src/object.rs

```diff
@@ -1436,12 +1438,13 @@ impl<T: ObjectType> ObjectExt for T {
 
     fn downgrade(&self) -> WeakRef<T> {
         unsafe {
-            let w = WeakRef(Box::new(mem::zeroed()), PhantomData);
+            let mut w = Box::new(mem::MaybeUninit::zeroed());
             gobject_sys::g_weak_ref_init(
-                mut_override(&*w.0),
+                mut_override((&mut *w).as_mut_ptr()),
                 self.as_object_ref().to_glib_none().0,
             );
-            w
+
+            WeakRef(Box::new((&mut *w).assume_init()), PhantomData)
```

Comment:
This creates two boxes and also changes the memory location of the weak reference, so it will break at runtime. It circumvents the whole reason why we box here at all.

The way how it was before is the best we can do here.